### PR TITLE
OPS-234: workaround for missing release branch deployments

### PR DESCRIPTION
**Pull request by booyaa**
### Overview 

Add a new function to use git to show commits for a given release. Azure DevOps is unable to show all commit histories for a release branch (this problem does not affect the "main" branch deploys).

I've created a test script so @andysbolton and TBC (someone who knows their way around git) can review

### How the script works

In essence we're performing the following: `git log --oneline --decorate --no-merges <last release branch>...<our release branch>`.

Most of the script involves working out what the last release branch is in the context of our release branch e.g. If our release branch is 5.35.0 then our last release branch will be 5..34.2
 
### Assumptions

you've got an up to date copy of `smartwyre-app` and it's in the same directory level as this repo. A

```powershell
cd smartwyre-app
..\azure-jira-update\TestScript.ps1 5.35.0  # view commits relating to 5.35.0
````

### CODE REVIEW

File: EP.PowerShell.JiraDeployInfo/Public/Update-AzureDeploymentInformation.ps1 [hunk 1]
```diff
@@ -116,11 +116,22 @@ function Update-AzureDeploymentInformation {
...
+        [string] $SourceBranch = "main",
+        
+        [Parameter(Mandatory=$false)][ValidateNotNullOrEmpty()]
+        [string] $TargetBranch = $env:BUILD_SOURCEBRANCHNAME
     )
 
     $jiraIds = @()
-    $jiraIds += (Get-JiraIDsFromAzureChanges -SystemAccessToken $SystemAccessToken -AzureChangeUrl $AzureChangeUrl)
+    if ($SourceBranch == "main") { # FIXME: hard coded
+        $jiraIds += (Get-JiraIDsFrmAzureChanges)
```
Comment (lsandoz) [inline EP.PowerShell.JiraDeployInfo/Public/Update-AzureDeploymentInformation.ps1:130]: Missing letter calling function, should be Get-JiraIDsFromAzureChanges

File: EP.PowerShell.JiraDeployInfo/Private/Get-JiraIDsFromGitLog.ps1 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    $jiraIDs = @()
+    git log --oneline $SourceCommit..."origin/$Branch" | ForEach-Object {
+        Find-JiraIDs ($_) | ForEach-Object {
+            $jiraIDs += $_.Value.ToUpper()
+        }
+    }
+
+    $jiraIDs | Sort-object | Get-Unique -AsString
+}
```
Comment (lsandoz) [inline EP.PowerShell.JiraDeployInfo/Private/Get-JiraIDsFromGitLog.ps1:19]: Shouldn't this one be the same as the one used on the testing script? Testing this one there didn't succeed to return what we expect. 

I don't know much of ps so maybe this is not that simple but I would try have the TestScript reference the real scripts used on the deploy as to mitigate possible discrepancies between the test version and the real one that could break the pipeline.

**Review by lsandoz — CHANGES_REQUESTED** [review]
Looks good overall, some small things I noticed. Will approve when fixed.

File: EP.PowerShell.JiraDeployInfo/Public/Update-AzureDeploymentInformation.ps1 [hunk 3]
```diff
@@ -116,11 +116,22 @@ function Update-AzureDeploymentInformation {
...
+        [Parameter(Mandatory=$false)][ValidateNotNullOrEmpty()]
+        [string] $SourceBranch = "main",
+        
+        [Parameter(Mandatory=$false)][ValidateNotNullOrEmpty()]
+        [string] $TargetBranch = $env:BUILD_SOURCEBRANCHNAME
     )
 
     $jiraIds = @()
-    $jiraIds += (Get-JiraIDsFromAzureChanges -SystemAccessToken $SystemAccessToken -AzureChangeUrl $AzureChangeUrl)
+    if ($SourceBranch == "main") { # FIXME: hard coded
```
Comment (andysbolton) [inline EP.PowerShell.JiraDeployInfo/Public/Update-AzureDeploymentInformation.ps1:129]: This will need to be `-eq` instead of `==`.

**Review by andysbolton — COMMENTED** [review]
The test script looks good and seems to work well. As @lsandoz said, the `EP.PowerShell.JiraDeployInfo/Private/Get-JiraIDsFromGitLog.ps1` function doesn't seem to correctly query for the previous release and returns different results, looks like that just wasn't updated.

File: EP.PowerShell.JiraDeployInfo/Public/Update-AzureDeploymentInformation.ps1 (hunk 1 above)
Comment (booyaa) [reply EP.PowerShell.JiraDeployInfo/Public/Update-AzureDeploymentInformation.ps1]: Good catch, not sure why the linter didn't pick it up! 🥇

**Review by lsandoz — APPROVED** [review]
Looks good to me now, nice work

**Pull request merged**